src/main/java/app/sami/languageWeb/extensionRequest/ExtensionRequestController.java

```diff
@@ -17,30 +24,52 @@ public class ExtensionRequestController {
 
     private final ExtensionRequestRepository extensionRequestRepository;
     private final ExtensionRequestService extensionRequestService;
+    private final ContractRepository contractRepository;
 
-    @PostMapping("/contracts/{contractId}/extension-request")
-    public ExtensionRequestDto createExtensionRequest(@PathVariable Long contractId,
+    @PostMapping("/extension-request")
+    public ExtensionRequestDto createExtensionRequest(
                                                       @RequestJwtSubject UUID subject,
-                                                      @RequestBody Long proposedDate){
+                                                      @RequestBody ExtensionRequestDto extensionRequestDto){
+        Contract contract = contractRepository.findById(extensionRequestDto.getContractId()).orElseThrow(NotFoundException::new);
+
+        if (!contract.isContracted(subject)){
+            throw new UserNotAllowedException("Only the contracted user can request for an extension.");
+        }
+        Optional<ExtensionRequest> extensionRequest = extensionRequestRepository.findByContractIdAndStatus(extensionRequestDto.getContractId(),
```

Comment:
this bit is unclear to me I have no idea why sometimes it is cancelled and sometimes not